lib/acts_as_list/active_record/acts/list.rb

```diff
@@ -480,6 +480,27 @@ def active_record_version_is?(version_requirement)
           version = Gem.loaded_specs['activerecord'].version
           requirement.satisfied_by?(version)
         end
+
+        def primary_key_query(id = nil)
+          id ||= send(self.class.primary_key)
+          ["#{quoted_table_name}.#{self.class.primary_key} != ?", id]
+        end
+
+        def composite_primary_key_query(id = nil)
+          keys = self.class.primary_key
+          if id
+            values = id
+          else
+            values = keys.map { |key| send(key) }
+          end
+
+          conditions = keys.map { |key| "#{quoted_table_name}.#{key} != ?" }.join(' AND ')
+          [conditions, *values]
+        end
+
+        def composite_primary_key?
+          self.class.respond_to?(:composite_primary_key?) && self.class.composite_primary_key?
+        end
```

Comment:
It strikes me that we could simplify this quite a bit by always making `primary_key` an array and then just mapping it out to a hash. Something like this?

```
def primary_key_condition(id = nil)
  primary_keys = Array.wrap(self.class.primary_key)

  id ? primary_keys.zip(id).to_h : slice(*primary_keys)
end
```

Ditch `composite_primary_key?` and then from the calling location do something like this instead:

```
where.not(primary_key_condition)
```

or

```
scope = scope.where.not(primary_key_condition(except.id))
```

And yes please add Rails 7.1 support to appraisal and the GitHub Actions matrix :)

In terms of the test suite, I think we also need some tests for scope changing just to be sure that works also. There might be a way to reuse existing basic tests but with composite ID's perhaps? https://github.com/brendon/acts_as_list/blob/master/test/shared_list.rb

Thanks again for the work on this :) It's given me some ideas for implementing it in my new positioning gem: https://github.com/brendon/positioning